### DISCUSSION TITLE
Double click to flash back under Ubuntu, but it can work normally with ". /" on the terminal

### DISCUSSION THREAD

**Discussion by FeiNoLabel** [pyinstaller/pyinstaller#6451] [Help]
On the Ubuntu terminal, use the ". /" program to run normally, but double-click the executable and right-click to run the program. The program will pop up but will flash back.
This program uses pyinstaller to package pyqt5. The program uses the target detection algorithm to detect objects in the environment circularly and display them on the interface made by pyqt5.

**Comment by bwoodsend** [comment]
That's probably either working directory dependence due to an improperly referenced resource or it's using subprocess to spawn an application which is added to `PATH` by a bashrc or some other shell initialisation file. You can check for the first just by running from another directory:
```
cd /tmp
/full/path/to/your/program
```
If that breaks, then you know it's the first. If it works fine then you'll need to find a way to capture whatever exception is being raised on double click. Since your using Qt, you might as well use a QDialog:
```python
import traceback
from PyQt5 import QtWidgets

app = QtWidgets.QApplication([])
try:
    ui = YourMainWindow()
    ui.show()
    app.exec()
except:
    QtWidgets.QMessageBox.critical(
        None, "Error",
        traceback.format_exc())
```
Hopefull, whatever this dialog says will help you find the real issue.
